syntax = "proto3";

package raft;

service Raft {
    rpc AppendEntries(AppendEntriesRequest) returns (AppendEntriesResponse) {}
    rpc RequestVote(RequestVoteRequest) returns (RequestVoteResponse) {}
}


message AppendEntriesRequest {
}

message AppendEntriesResponse {
}

message RequestVoteRequest {
    int32 term = 1;
    string candidate_id = 2;
    int32 last_log_index = 3;
    int32 last_log_term = 4;
}

message RequestVoteResponse {
    int32 term = 1;
    bool granted = 2;
}

message LogEntry {
    int32 term = 1;
    string command = 2;
}
